### PR TITLE
Colors - adjust palette to steps of 9 when addDarkestTint is true

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
Colors - adjust palette to steps of 9 when addDarkestTint is true

## Changelog
Colors - adjust palette to steps of 9 when addDarkestTint is true

## Additional info
WOAUILIB-3830

### CODE REVIEW

File: src/style/__tests__/colors.spec.js [hunk 1]
```diff
@@ -112,38 +116,106 @@ describe('style/Colors', () => {
...
+      expect(palette.length).toBe(8);
+      expect(palette).toContain(baseColor);
+      expect(palette).toEqual(tints);
     });
 
-    it('should generateColorPalette with adjusted saturation', () => {
-      const palette = uut.generateColorPalette('#FFE5FF');
-      expect(palette).toEqual(['#661A66', '#8F248F', '#B82EB7', '#D148D1', '#DB71DB', '#E699E6', '#F0C2F0', '#FFE5FF']);
+    it('should generateColorPalette with adjustLightness option true (default)', () => {
+      const palette = uut.generateColorPalette(baseColor, {adjustLightness: true});
```
Comment (M-i-k-e-l) [inline src/style/__tests__/colors.spec.js:147]: This is a little odd IMO, you send the default value and test it, shouldn't the test above be enough for it?

File: src/style/__tests__/colors.spec.js [hunk 2]
```diff
@@ -112,38 +116,106 @@ describe('style/Colors', () => {
   });
 
   describe('generateColorPalette', () => {
+    const baseColor = '#3F88C5';
+    const tints = ['#193852', '#255379', '#316EA1', '#3F88C5', '#66A0D1', '#8DB9DD', '#B5D1E9', '#DCE9F4'];
+    const baseColorLight = '#DCE9F4';
+    const tintsLight = ['#1A3851', '#265278', '#326D9F', '#4187C3', '#68A0CF', '#8EB8DC', '#B5D1E8', '#DCE9F4'];
+    const tintsAdjustedLightness = ['#193852', '#255379', '#316EA1', '#3F88C5', '#66A0D1', '#8DB9DD', '#B5D1E9', '#DCE9F4'];
```
Comment (M-i-k-e-l) [inline src/style/__tests__/colors.spec.js:123]: It looks like there is duplication of consts, why do we need them?
`tintsAdjustedLightness` = `tints`
`tintsAdjustedSaturation` = `tintsLight`

File: src/style/colors.ts [hunk 3]
```diff
@@ -323,7 +337,31 @@ function colorStringValue(color: string | object) {
   return color?.toString();
 }
 
-function adjustSaturation(colors: string[], color: string) {
+function adjustAllSaturations(colors: string[], color: string, levels: number[]) {
```
Comment (M-i-k-e-l) [inline src/style/colors.ts:340]: is `color: string` actually `selectedColor`\`currentColor`?

File: src/style/colors.ts [hunk 4]
```diff
@@ -323,7 +337,31 @@ function colorStringValue(color: string | object) {
...
+  const array: string[] = [];
+  _.forEach(colors, (c, index) => {
+    if (c === color) {
+      array[index] = color;
+    } else {
+      const hsl = Color(c).hsl();
+      const saturation = hsl.color[1];
+      const level = levels[index];
+      if (level) {
```
Comment (M-i-k-e-l) [inline src/style/colors.ts:349]: This will ignore `0`

File: src/style/colors.ts [hunk 5]
```diff
@@ -323,7 +337,31 @@ function colorStringValue(color: string | object) {
...
+        const saturationLevel = saturation + level;
+        const clampedLevel = _.clamp(saturationLevel, 0, 100);
+        const adjusted = addSaturation(c, clampedLevel);
```
Comment (M-i-k-e-l) [inline src/style/colors.ts:352]: Maybe it's out of scope but I think `addSaturation` is actually `setSaturation`

File: demo/src/screens/foundationScreens/ColorsScreen.js [hunk 6]
```diff
@@ -281,11 +291,99 @@ class ColorsScreen extends Component {
...
+      </>
+    );
+  };
+
   render() {
     return (
       <>
         {this.renderSearch()}
         <ScrollView ref={this.scrollViewRef}>
+          {/* {this.renderColorPaletteExample()} */}
```
Comment (M-i-k-e-l) [inline demo/src/screens/foundationScreens/ColorsScreen.js:386]: Not sure why it's hidden, wouldn't UX will want to test it?
BTW, I did not test it.
We can go over it together if you like

**Review by M-i-k-e-l — COMMENTED** [review]
PLEASE try to refactor in separate commits 🙏 
I think it would have been much easier to look at.

File: src/style/__tests__/colors.spec.js (hunk 1 above)
Comment (Inbal-Tish) [reply src/style/__tests__/colors.spec.js:147]: Right, I wanted to explicitlly test it but it is redundent

File: src/style/colors.ts (hunk 3 above)
Comment (Inbal-Tish) [reply src/style/colors.ts:340]: Yes... chenged to 'baseColor' to be extra clear

File: src/style/colors.ts (hunk 4 above)
Comment (Inbal-Tish) [reply src/style/colors.ts:349]: That's fine... Should it be explicite?

File: src/style/colors.ts (hunk 5 above)
Comment (Inbal-Tish) [reply src/style/colors.ts:352]: Agree

File: demo/src/screens/foundationScreens/ColorsScreen.js (hunk 6 above)
Comment (Inbal-Tish) [reply demo/src/screens/foundationScreens/ColorsScreen.js]: Moved to the bottom of the screen

**Review by M-i-k-e-l — APPROVED** [review]

**Pull request merged**